pyccel/parser/semantic.py

```diff
@@ -2697,6 +2697,8 @@ def _handle_is_operator(self, IsClass, expr, **settings):
             var1, var2 = var2, var1
 
         if isinstance(var2, Nil):
+            if isinstance(var1, Nil):
+                return IsClass(expr.lhs, expr.rhs)
```

Comment:
This will create `IsClass(Nil,Nil)`. Is that done on purpose? If yes then how is it evaluated

Reply:
the None comparison is implemented only in fortran, so i decided to raise the error in the fortran code generation stage.

Reply:
Ok. When the None comparison is implemented in C it will be in a way that will allow it to be symmetrical so this makes sense